qt_ui/windows/mission/flight/waypoints/QFlightWaypointList.py

```diff
@@ -47,23 +56,39 @@ def update_list(self):
             total_column_width += self.columnWidth(i) + self.lineWidth()
         self.setFixedWidth(total_column_width)
 
-    def add_waypoint_row(
+        # reconnect itemChanged signal to handler slot for making altitude changes
+        self.model.itemChanged.connect(self.on_changed)
+
+    def _add_waypoint_row(
         self, row: int, flight: Flight, waypoint: FlightWaypoint
     ) -> None:
         self.model.insertRow(self.model.rowCount())
 
         self.model.setItem(row, 0, QWaypointItem(waypoint, row))
 
         altitude = int(waypoint.alt.feet)
-        altitude_type = "AGL" if waypoint.alt_type == "RADIO" else "MSL"
-        altitude_item = QStandardItem(f"{altitude} ft {altitude_type}")
-        altitude_item.setEditable(False)
+        altitude_item = QStandardItem(f"{altitude}")
+        altitude_item.setEditable(True)
         self.model.setItem(row, 1, altitude_item)
 
+        altitude_type = "AGL" if waypoint.alt_type == "RADIO" else "MSL"
+        altitude_type_item = QStandardItem(f"{altitude_type}")
+        altitude_type_item.setEditable(False)
+        self.model.setItem(row, 2, altitude_type_item)
+
         tot = self.tot_text(flight, waypoint)
         tot_item = QStandardItem(tot)
         tot_item.setEditable(False)
-        self.model.setItem(row, 2, tot_item)
+        self.model.setItem(row, 3, tot_item)
+
+    def on_changed(self) -> None:
+        for i in range(self.model.rowCount()):
+            altitude = self.model.item(i, 1).text()
+            try:
+                altitude_feet = float(altitude)
+            except:
```

Comment:
No bare except (this will also catch `SystemExit` and `KeyboardInterrupt`, and who knows what else that shouldn't be ignored).

It looks like you're trying it ignore `ValueError`? I'm not really sure why though. Catch `ValueError` explicitly and add a comment explaining the circumstances?

Reply:
Yep, I was trying to ignore ValueError. The idea is that if the user types in an invalid number that can't be cast to a float, it will be ignored.

Reply:
Ah. Make it a spin box instead of a text box so invalid input isn't possible?